src/MacroTools/Buffs/AddSpellonCastBuff.cs

```diff
@@ -0,0 +1,36 @@
+﻿using System.Collections.Generic;
+using WCSharp.Buffs;
+using static War3Api.Common;
+
+namespace MacroTools.Buffs
+{
+  public sealed class AddSpellonCastBuff : PassiveBuff
+  {
+    public IEnumerable<int>? AbilitiesToAdd { get; init; }
+    
+    public override void OnApply()
+    {
+      if (AbilitiesToAdd != null)
+        foreach (var ability in AbilitiesToAdd)
+        {
+          SetPlayerAbilityAvailable(GetOwningPlayer(Caster), ability, true);
```

Comment:
Let's just get rid of this line, and the corresponding one in `OnDispose`. This stops the Buff from working coherently on an arbitrary number of units.

Reply:
not sure what you mean here

Reply:
> not sure what you mean here

Which part?

Reply:
all of it? i dont understand what you are telling me

Reply:
So right now, the Buff runs a function that applies to the _player_ with the buff. That means the behaviour is unpredictable whenever you apply it to more than one unit. We shouldn't have that in a generic buff. So the solution is to remove both of the `SetPlayerAbilityAvailable` lines, which aren't necessary to the function of the spell anyway since you're adding and removing the ability dynamically.